_posts/2019-05-21-i18n-messages.adoc

```diff
@@ -0,0 +1,327 @@
+---
+compatible_cuba_versions: 7.0+
+compatible_java_versions: 8+
+project_id: cuba-petclinic-i18n-messages
+permalink: localization-in-cuba-applications
+---
+= Localization in CUBA applications
+:showtitle:
+:sectlinks:
+:sectanchors:
+:page-navtitle: Localization in CUBA applications
+:page-excerpt: In this guide the localization features of a CUBA application are explained
+:page-root: ../../../
+:project_id: cuba-petclinic-i18n-messages
+:java_version: 1.8
+:cuba_version: 7.0
+
+
+CUBA applications have build-in support for localized messages for different languages. Translations are externalized in
+properties files to have a clear separation between the application code and the translated messages.
+
+
+== What Will be Built
+
+This guide enhances the https://github.com/cuba-platform/cuba-petclinic[CUBA petclinic] example to show how localized messages within the CUBA application can be defined and used:
+
+* the application is also available in German and Russian language
+* dynamic data as automatically generated descriptions for visits are localized
+* provide german translation for User browse platform screen
```

Comment:
This statement should be rephrased if the whole German translation will be applied.